backend/src/data/repositories/master/master.repository.ts

```diff
@@ -17,18 +21,32 @@ class Master {
     return masters.map(Master.modelToEntity);
   }
 
-  create(master: MasterEntity): Promise<void> {
-    return this.#MasterModel
+  async getByFilter(
+    filterName: Filter,
+    value: string,
+  ): Promise<MasterEntity | undefined> {
+    const master = await this.#MasterModel
       .query()
-      .insert({
-        id: master.id,
-        email: master.email,
-        name: master.name,
-        passwordHash: '', //TODO: replace by actual values
-        passwordSalt: '',
-        createdAt: master.createdAt.toISOString(),
-      })
-      .then();
+      .select()
+      .where({ [filterName]: value })
+      .first();
+    if (!master) {
+      return;
+    }
+
+    return Master.modelToEntity(master);
+  }
+
+  async create(password: string, master: MasterEntity): Promise<MasterM> {
+    const salt = createSalt(saltRounds);
+    return this.#MasterModel.query().insert({
+      id: master.id,
+      email: master.email,
+      name: master.name,
+      passwordHash: await encrypt(password, salt),
```

Comment:
Repository should not contain any logic except database querying/inserting. Entity you pass to repository should contain all required info